### DISCUSSION TITLE
Azure as provider working?

### DISCUSSION THREAD

**Discussion by ioucyf** [supabase/supabase#12065] [Questions]
I've spent more than two hours just trying to figure out how to get it to work. I have followed the guide in the docs as instructed. 

Has there been updates to Azure that broke support? 

I can't figure out the problem.

url returned: http://localhost:5173/login?error=server_error&error_description=Error+getting+user+email+from+external+provider

**Comment by silentworks** [comment]
I ran into this same issue and it turns out that Microsoft does some messy saving of user's email at times and causes this issue. You will need to use your user account and log into the Azure portal https://portal.azure.com/ to update your profile and add your email address to the email field inside of the portal. This should now allow you to `signInWithOAuth` with the `scopes: 'email'` and have that working.
